web/src/topic_list.ts

```diff
@@ -347,6 +347,10 @@ export function initialize({
                 return;
             }
 
+            if ($(e.target).hasClass("zulip-icon-follow")) {
+                return;
+            }
```

Comment:
Let's add a proper class naming this element, `left-sidebar-toggle-visibility-policy`, rather than writing logic based on which icon is present.